config/routes.rb

```diff
@@ -0,0 +1,17 @@
+Rails.application.routes.draw do
+  devise_for :users
+  resources :spendings
+  resources :types
+  resources :users
```

Comment:
- [ ] Please make sure to use only the necessary routes for your project. I also checked your controllers, and you have some actions like `edit` or `show` on some controllers that are currently empty, so it would be a good idea to delete those ones that you are not using.